frontend/src/screens/ProjectEditor.tsx

```diff
@@ -1,162 +1,1261 @@
         e.preventDefault();
-        save();
+        saveCurrentFile();
       }
     };
     window.addEventListener("keydown", handleKeyDown);
     return () => window.removeEventListener("keydown", handleKeyDown);
   }, [userIsOwner, projectFiles, projectName, projectVersion]);
 
-  const copy = () => {
-    window.location.href = `/c/${projectName}`;
+  // honestly idk this barely works
+  const [paneWidths, setPaneWidths] = React.useState(() => ({
+    ...DEFAULT_PANE_WIDTHS
+  }));
+  const paneOrder = paneState.order.filter((p) => paneState.open[p]);
+
+  const closedPanes = paneTypes.filter((p) => !paneState.open[p.key]);
+  // const openPanes = paneState.order.filter((p) => paneState.open[p]);
+
+  const resizingRef = React.useRef<{
+    idx: number;
+    startX: number;
+    startWidths: number[];
+  } | null>(null);
+
+  const onResizerMouseDown = (idx: number, e: React.MouseEvent) => {
+    resizingRef.current = {
+      idx,
+      startX: e.clientX,
+      startWidths: paneOrder.map((p) => paneWidths[p] || DEFAULT_PANE_WIDTHS[p])
+    };
+    document.addEventListener("mousemove", onResizerMouseMove);
+    document.addEventListener("mouseup", onResizerMouseUp);
   };
 
-  const handleFileEdit = (newContent: string) => {
-    const updatedFiles: ProjectFile[] = projectFiles.map(
-      (file: ProjectFile) => {
-        if (file.fileName === currentlySelectedFilename) {
-          return {
-            fileName: currentlySelectedFilename,
-            fileContent: newContent
-          };
-        }
+  const onResizerMouseMove = (e: MouseEvent) => {
+    if (!resizingRef.current) return;
+    const { idx, startX, startWidths } = resizingRef.current;
+    const delta = e.clientX - startX;
+    const leftPane = paneOrder[idx];
+    const rightPane = paneOrder[idx + 1];
 
-        return file;
-      }
-    );
+    const totalWidth = startWidths[idx] + startWidths[idx + 1];
 
-    setProjectFiles(updatedFiles);
+    let newLeft = startWidths[idx] + delta;
+    let newRight = startWidths[idx + 1] - delta;
+
+    if (newLeft < MIN_PANE_WIDTH) {
+      newLeft = MIN_PANE_WIDTH;
+      newRight = totalWidth - MIN_PANE_WIDTH;
+    } else if (newRight < MIN_PANE_WIDTH) {
+      newRight = MIN_PANE_WIDTH;
+      newLeft = totalWidth - MIN_PANE_WIDTH;
+    }
+
+    setPaneWidths((prev) => ({
+      ...prev,
+      [leftPane]: newLeft,
+      [rightPane]: newRight
+    }));
   };
 
-  if (projectData.isLoading) return <p>Loading...</p>;
-  if (projectData.error) return <p>Error loading project</p>;
+  const onResizerMouseUp = () => {
+    document.removeEventListener("mousemove", onResizerMouseMove);
+    document.removeEventListener("mouseup", onResizerMouseUp);
+    resizingRef.current = null;
+  };
 
-  return (
-    <div
+  const renderPane = (pane: string, width?: number) => {
+    if (!paneState.open[pane]) return null;
+    switch (pane) {
+      case "explorer":
+        return (
+          <Paper

```

Comment:
cannot imagine a better use case for components!!!!!

imagine just having `<ExplorerPane />` right here